test/test_index.py

```diff
@@ -1218,6 +1218,31 @@ def test_index_add_non_normalized_path(self, rw_repo):
 
         rw_repo.index.add(non_normalized_path)
 
+    @with_rw_directory
+    def test_index_version_v3(self, tmp_dir):
+        tmp_dir = Path(tmp_dir)
+        with cwd(tmp_dir):
+            subprocess.run(["git", "init", "-q"], check=True)
+            file = tmp_dir / "file.txt"
+            file.write_text("hello")
+            subprocess.run(["git", "add", "-N", "file.txt"], check=True)
+
+            repo = Repo(tmp_dir)
+
+            assert len(repo.index.entries) == 1
```

Comment:
To test this, I think the `index.version` must be exposed and validated here.
I don't see at all why it would use index version 3 here.